libsolidity/ast/ASTJsonConverter.cpp

```diff
@@ -704,13 +704,13 @@ bool ASTJsonConverter::visit(ElementaryTypeNameExpression const& _node)
 
 bool ASTJsonConverter::visit(Literal const& _node)
 {
-	char const* tokenString = Token::toString(_node.token());
+	string tokenString = literalTokenKind(_node.token());
 	Json::Value value{_node.value()};
 	if (!dev::validateUTF8(_node.value()))
 		value = Json::nullValue;
 	Token::Value subdenomination = Token::Value(_node.subDenomination());
 	std::vector<pair<string, Json::Value>> attributes = {
-		make_pair("token", tokenString ? tokenString : Json::Value()),
+		make_pair(m_legacy ? "token" : "kind", tokenString),
```

Comment:
If we want to keep backwards compatibility (if it is worth anything in this case), `token` shouldn't point to kind. 

The only difference is that in case of a boolean `token` would be `true` or `false`, while it is `bool` in the new way.

I'd vote just to ignore this and keep your change as is.